From 12860bde0ad817816d68c7e37940500fb1f2d349 Mon Sep 17 00:00:00 2001
From: Hongcheng Xie <hongcheng.xie@intel.com>
Date: Tue, 25 Jan 2022 10:47:35 +0800
Subject: [PATCH] Remount proc with android needed option

Docker container will mount proc when container running,
instead mount proc again in init, remount proc with android
needed option((hidepid=2,gid=3009).

Tracked-On:OAM-100816
Change-Id: Ic8c09475279ea4320832b2f881cd94e5efc2a4eg
Signed-off-by: Hongcheng Xie <hongcheng.xie@intel.com>
---
 init/first_stage_init.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/init/first_stage_init.cpp b/init/first_stage_init.cpp
index c237f027a..716129338 100644
--- a/init/first_stage_init.cpp
+++ b/init/first_stage_init.cpp
@@ -203,7 +203,7 @@ int FirstStageMain(int argc, char** argv) {
     coldbootopt_main("/coldboot_dev.json", find_container_id(argc, argv));
 #endif
 #define MAKE_STR(x) __STRING(x)
-    CHECKCALL(mount("proc", "/proc", "proc", 0, "hidepid=2,gid=" MAKE_STR(AID_READPROC)));
+    mount("proc", "/proc", "proc", MS_REMOUNT, "hidepid=2,gid=" MAKE_STR(AID_READPROC));
 #undef MAKE_STR
     // Don't expose the raw commandline to unprivileged processes.
     CHECKCALL(chmod("/proc/cmdline", 0440));
-- 
2.39.0

